### DISCUSSION TITLE
Seeking for guidance - commit to repoA creates a branch from RepoB and makes a PR to RepoB

### DISCUSSION THREAD

**Discussion by jasontsaitks** [community/community#26085] [Actions]
<p>Hi,</p>
<p>I am new to Github Action and looking for guidance. We have a requirement:</p>
<p>Repo A<br>
Repo B</p>
<p>repoA stores a template file<br>
repoB stores artifacts generated from repoA’s template file</p>
<p>committing to Repo A shall trigger action that:</p>
<ol>
<li>do some work and generate a artifacts (some files)</li>
<li>repoB stores the artifacts. RepoA’s newly generated artifacts shall be in a branch in repoB and PR to repoB’s main branch</li>
</ol>
<p>Could I have some guidance if this is possible and how to start?</p>
<p>Thanks.</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group quote-modified" data-username="jasontsaitks" data-post="1" data-topic="127206">
<div class="title">
<div class="quote-controls"></div>
 jasontsaitks:</div>
<blockquote>
<p>committing to Repo A shall create a branch from Repo B</p>
</blockquote>
</aside>
<p>Could you clarify what you mean by this? Do you want to automatically mirror branches from A to B, without any pull request?</p>

**Comment by jasontsaitks** [comment]
<p>thanks for the reply.</p>
<p>repoA stores a template file<br>
repoB stores artifacts generated from repoA’s template file</p>
<p>Repo A’s action shall</p>
<ol>
<li>do some work and generate a artifacts (some files)</li>
<li>repoB stores the artifacts. RepoA’s newly generated artifacts shall be in a branch in repoB and PR to repoB’s main branch</li>
</ol>

**Comment by peter-evans** [comment, marked as answer]
<p>This is a rough outline of how you could design the workflow. Note that you need to use a <code>repo</code> scoped <a href="https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token" rel="nofollow noopener">Personal Access Token</a> to create the pull request in a remote repository.</p>
<p>This workflow goes in repo-a:</p>
<pre><code class="lang-auto">on:
  push:
    branches: [master]
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      # Checkout repo-a
      - uses: actions/checkout@v2

      # Steps that use the template to build artifacts go here

      # Save the artifact
      - uses: actions/upload-artifact@v2
        with:
          name: artifact
          path: path/to/artifact

      # Checkout the default branch of repo-b
      - uses: actions/checkout@v2
        with:
          token: ${{ secrets.PAT }}
          repository: owner/repo-b

      # Download the artifact to the correct path
      - uses: actions/download-artifact@v2
        with:
          name: artifact
          path: path/to/download/artifact

      # Raise a pull request from artifact-update to the default branch
      - name: Create Pull Request
        uses: peter-evans/create-pull-request@v3
        with:
          token: ${{ secrets.PAT }}
          branch: artifact-update

</code></pre>
<p>If you prefer, instead of using <code>upload-artifact</code> and <code>download-artifact</code> you can just checkout repo-b to a different path that doesn’t overwrite repo-a and then copy the artifact to the right place. If you do that, make sure to use the <code>path</code> input of <code>create-pull-request</code> action so it’s pointing to where repo-b is checked out.</p>

**Comment by jasontsaitks** [comment]
<p>thank you. It is so detailed.<br>
I am going to try now.<br>
Thank you again.</p>

**Comment by jasontsaitks** [comment]
<p>Now I am thinking again…I wonder if I shall checkout repoB at the first place since repoA generates the new artifact and it will be sent to repoB in PR?</p>
<p>I am considering a scenario where:</p>
<ol>
<li>repoB has: A B C</li>
<li>now repoA generates A B D</li>
<li>if I download upload/download artifacts, won’t my path now contains: A B C D, but I just want repoB’s latest master to contain only: A B D</li>
</ol>
<p>Thanks</p>

**Comment by airtower-luna** [comment]
<p>You will need to checkout repo B to work with it. If it contains artifacts you don’t want any more you’ll have to delete them, like when working with git locally. If you want <em>only</em> the files from the artifact you can simply do something like</p>
<pre><code class="lang-yaml">- name: Delete old artifacts
  run: |
    rm path/to/download/artifact/*
</code></pre>
<p>to delete all of them. I’m not familiar with the <code>peter-evans/create-pull-request</code> action, so I don’t know if it’ll pick up deletion automatically, or if you’ll need to use <code>git rm</code>.</p>

**Comment by jasontsaitks** [comment]
<p>thank you for the reply. I end up:</p>
<ol>
<li>create a folder under repoB current_artifact</li>
</ol>
<p>In the workflow:</p>
<ol>
<li>repoA generate new artifact to new_artifact</li>
<li>checkout repoB</li>
<li>rsync  new_artifact current_artifact</li>
<li>PR to repoB</li>
</ol>
